internal/loader/loader.go

```diff
@@ -254,10 +269,19 @@ func loadFromStruct(f *ast.File, name string, info *types.Info) (*Result, error)
 		for _, l := range fn.Recv.List {
 			var ident *ast.Ident
 			switch t := l.Type.(type) {
-			case *ast.StarExpr:
-				ident = t.X.(*ast.Ident)
 			case *ast.Ident:
 				ident = t
+			case *ast.StarExpr:
+				switch t := t.X.(type) {
+				case *ast.Ident:
+					ident = t
+				case *ast.IndexExpr:
+					// single type parameter
+					ident = t.X.(*ast.Ident)
+				case *ast.IndexListExpr:
+					// multiple type parameters
+					ident = t.X.(*ast.Ident)
```

Comment:
is handling both `*ast.IndexListExp` and `*ast.IndexExpr` cases meant to be indentical?

Reply:
yep - one is analog to `MyType[T any]` and the other is `MyType[T any, U any, V any]`... single type arg vs multiple.